nebula-logger/core/main/logger-engine/classes/Logger.cls

```diff
@@ -2553,22 +2588,27 @@ global with sharing class Logger {
         // This gives us a chance to run the handler class & handler plugins before insert,
         // allowing the plugins to make further changes to the `LogEntryEvent__e` records
         // So, execute the handler, which internally then executes any plugins
-        new LogEntryEventHandler().executeBeforeInsert(logEntryEvents);
+        LoggerTriggerableContext logEntryEventsBeforeContext = new LoggerTriggerableContext(
+            Schema.LogEntryEvent__e.SObjectType,
+            TriggerOperation.BEFORE_INSERT,
+            logEntryEvents
+        );
+        LoggerSObjectHandler.getHandler(Schema.LogEntryEvent__e.SObjectType)?.overrideTriggerableContext(logEntryEventsBeforeContext).execute();
 
         // Now that the plugins have run, double check to make sure that saving should still happen
         if (getBufferSize() == 0) {
             return;
         }
 
-        switch on saveMethod {
-            when EVENT_BUS {
-                List<Database.SaveResult> saveResults = EventBus.publish(logEntryEvents);
-                LoggerEmailUtils.sendErrorEmail(Schema.LogEntryEvent__e.SObjectType, saveResults);
+        switch on saveMethodName {
```

Comment:
couldn't you call `SaveMethod.valueOf` here and keep the enum switch statement?

Reply:
Yeah, I think I can use `SaveMethod.valueOf` to avoid the hardcoded strings - [related to my other comment](https://github.com/jongpie/NebulaLogger/pull/298#discussion_r859322016), I'll have do a try/catch to get the enum value here, since plugins can now introduce new string values for save methods, which won't have an equivalent enum value. But that still sounds better than hardcoding strings here!

Reply:
awww yeah